### PR TITLE
[refactor] [new] Event system

### PR DESCRIPTION
**Pull request by Xiaro**
**Describe the pull**
1. Added our own event system
2. Removed Alpine from dependencies
3. Marked `onUpdate()`, `onRender()`, `onWorldRender()` as deprecated

**Describe how this pull is helpful**
1. Much cleaner syntax for Listener
2. Less dependencies
3. see #1432 

**Additional context**
Add any other context about the pull here.

### CODE REVIEW

File: src/main/java/me/zeroeightsix/kami/event/ForgeEventProcessor.kt [hunk 1]
```diff
@@ -33,48 +33,63 @@ object ForgeEventProcessor {
...
+                KamiEventBus.post(ResolutionUpdateEvent(mc.displayWidth, mc.displayHeight))
                 for (component in KamiMod.getInstance().guiManager.children) {
                     if (component !is Frame) continue
                     KamiGUI.dock(component)
                 }
             }
-        }
-
-        if (mc.world != null && mc.player != null) {
-            ModuleManager.onUpdate()
```
Comment (l1ving) [inline src/main/java/me/zeroeightsix/kami/event/ForgeEventProcessor.kt:52]: that moment when onUpdate() was being called twice :moyai:

Comment (Xiaro) [reply src/main/java/me/zeroeightsix/kami/event/ForgeEventProcessor.kt:52]: That is deletion 🐒

**Review by l1ving — CHANGES_REQUESTED** [review]

File: src/main/java/me/zeroeightsix/kami/module/Module.kt [hunk 2]
```diff
@@ -129,9 +135,16 @@ open class Module {
...
+    /** Use event listener for [SafeTickEvent] Instead */
+    @Deprecated ("Planned to remove by 1.1.8 release")
+    open fun onUpdate(event: SafeTickEvent) {}
+    /** Use event listener for [RenderOverlayEvent] Instead */
+    @Deprecated ("Planned to remove by 1.1.8 release")
     open fun onRender() {}
-    open fun onWorldRender(event: RenderEvent) {}
+    /** Use event listener for [RenderOverlayEvent] Instead */
+    @Deprecated ("Planned to remove by 1.1.8 release")
+    open fun onWorldRender(event: RenderWorldEvent) {}
```
[the remark this answers is not in the document]
Comment (Xiaro) [reply src/main/java/me/zeroeightsix/kami/module/Module.kt:146]: 1.1.8 release != 1.1.8 beta

[the remark this answers is not in the document]
Comment (l1ving) [reply src/main/java/me/zeroeightsix/kami/module/Module.kt:146]: Doesn't matter, there will be no 1.1.8 beta

File: src/main/java/me/zeroeightsix/kami/module/ModuleManager.kt [hunk 3]
```diff
@@ -97,23 +99,16 @@ object ModuleManager {
...
             if (isModuleListening(module)) {
                 KamiTessellator.prepareGL()
-                module.onWorldRender(e)
+                module.onWorldRender(event)
                 KamiTessellator.releaseGL()
             }
         }
-        GlStateManager.glLineWidth(1f)
-        KamiTessellator.releaseGL()
-        mc.profiler.endSection()
```
[the remark this answers is not in the document]
Comment (Xiaro) [reply src/main/java/me/zeroeightsix/kami/module/ModuleManager.kt:116]: `KamiTessellator.releaseGL()` does it already

**Review by l1ving — APPROVED** [review]

**Pull request merged**